kaltura-ui/src/details-bar/details-bar.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Output, EventEmitter, Component, ElementRef, Input, ViewEncapsulation, AfterViewInit, ViewChild, HostListener } from '@angular/core';
+import { DetailInfoComponent } from './detail-info.component';
+
+export interface DetailInfo {
+  caption?: string;
+  value?: string;
+  link?: string;
+  tooltip?: string;
+  tooltipAsHTML?: boolean;
+  iconStyle?: string;
+  itemStyle?: string;
+}
+
+@Component({
+  selector: 'k-details-bar',
+  templateUrl: './details-bar.component.html',
+  styleUrls: ['./details-bar.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class DetailsBarComponent implements AfterViewInit {
+
+  @Input() data: DetailInfo[] = [];
+  @Input() captionField: string;
+  @Input() valueField: string;
+  @Input() linkField: string;
+  @Input() iconStyleField: string;
+  @Input() tooltipField: string;
+  @Input() separatorField: string = "|";
+  @Input() maxItemWidthField: number = 100;
+
+  @ViewChild('dataPanel') dataPanel: ElementRef;
+  @ViewChild('dataWrapper') dataWrapper: ElementRef;
+  public _showMore: boolean = false;
+  private showMoreCheckIntervalID: number;
+  public _showMoreEnabled: boolean = true;
+
+  public text: string;
+
+  ngAfterViewInit() {
+    this.checkShowMore();
+  }
+
+  @HostListener('window:resize')
+  private onResize() {
+    this.checkShowMore();
+  }
+
+  checkShowMore() {
+    // we use a cancelable interval to improve performances on window resize
+    if (this.showMoreCheckIntervalID) {
+      clearInterval(this.showMoreCheckIntervalID);
+      this.showMoreCheckIntervalID = null;
+    }
+    this.showMoreCheckIntervalID = setTimeout(() => {
```

Comment:
Interval, but you use setTimeout. 
Either way, maybe debouncing is the way to go.